SU-SWS/stanford_sites_helper
# SITES-206: Favor SSO authentication over local

# READY FOR REVIEW

# Summary
- Favor SSO authentication over local authentication.
- Favor SAML over WebAuth
- Allow Webauth if no SAML
- Allow local if no SAML nor WebAuth

# Needed By (Date)
- When does this need to be merged by? Socorro go-live

# Criticality
- How critical is this PR on a 1-10 scale? 9
- It has security and UX implications for going live with new sites on ACSF

# Steps to Test

1. Check out this branch
2. Install {{stanford_ssp}}
3. Go to the homepage
4. Ensure that "Log in with your SUNetID" links to `sso/login`
5. Disable {{stanford_ssp}} and {{stanford_simplesamlphp_auth}}
6. Enable {{webauth}}
7. Go to the homepage
8. Ensure that "Log in with your SUNetID" links to `sites/default/webauth/login`

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules? Sites 2.0/ACSF

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SITES-206

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
